pkg/noderesourcetopology/filter.go

```diff
@@ -44,23 +46,46 @@ func singleNUMAContainerLevelHandler(pod *v1.Pod, zones topologyv1alpha1.ZoneLis
 	// Node() != nil already verified in Filter(), which is the only public entry point
 	logNumaNodes("container handler NUMA resources", nodeInfo.Node().Name, nodes)
 
-	// We count here in the way TopologyManager is doing it, IOW we put InitContainers
-	// and normal containers in the one scope
-	for _, container := range append(pod.Spec.InitContainers, pod.Spec.Containers...) {
+	// the init containers are running SERIALLY and BEFORE the normal containers
+	// therefore, we don't need to accumulate their resources together
+	for _, initContainer := range pod.Spec.InitContainers {
+		logKey := fmt.Sprintf("%s/%s/%s", pod.Namespace, pod.Name, initContainer.Name)
+		klog.V(6).InfoS("target resources", resourceListToLoggable(logKey, initContainer.Resources.Requests)...)
+
+		_, match := resourcesAvailableInAnyNUMANodes(logKey, nodes, initContainer.Resources.Requests, qos, nodeInfo)
+		if !match {
+			// we can't align init container, so definitely we can't align a pod
+			return framework.NewStatus(framework.Unschedulable, fmt.Sprintf("cannot align init container: %s", initContainer.Name))
+		}
+	}
+
+	// this flow is changing the values of zones for its internal accounting.
+	// hence, create a pristine copy for this purpose and keep the original intact
+	mutableZones := zones.DeepCopy()
```

Comment:
I was wondering that maybe extending `NUMANode` to track the currently available resources can lead to nicer code, and most imporant can save us to call `createNUMANodeList` for each container
```
type NUMANode struct {
	NUMAID    int
	Resources v1.ResourceList
        Available v1.ResourceList
}
```

Reply:
I see the point indeed. But its might be a bit confusing since the ```NUMANode.Resources``` value is already taking the ```Available``` resources from the ```Zone```: https://github.com/kubernetes-sigs/scheduler-plugins/blob/7a6dcdc99b1ee9a324823eaf98718cfd9e98e805/pkg/noderesourcetopology/pluginhelpers.go#L149

I think it will be better subtracting the resources from the ```NUMANode``` object directly instead from the ```Zone``` and then recreating the ```NUMANode```

Reply:
good point. let's postpone this work as further optimization.

Reply:
The change wasn't big so I applied it already.